source/include/shadow.h

```diff
@@ -217,119 +245,153 @@ typedef enum ShadowStatus
  *
- * When thingName is only known at run time and held in a variable myThingName, invoke
- * the macro like this:
+ * When thingName and shadowName are only known at run time, and held in variables myThingName
+ * and myShadowName, invoke the macro like this:
  *
  *     SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_LENGTH, SHADOW_SUFFIX_DELTA_LENGTH,
- *                       strlen( ( const char * ) myThingName ) )
- *
- * @param[operationLength] Can be one of:
- *                          - #SHADOW_OP_UPDATE_LENGTH
- *                          - #SHADOW_OP_DELETE_LENGTH
- *                          - #SHADOW_OP_GET_LENGTH
- * @param[suffixLength]    Can be one of:
- *                          - #SHADOW_SUFFIX_NULL_LENGTH
- *                          - #SHADOW_SUFFIX_ACCEPTED_LENGTH
- *                          - #SHADOW_SUFFIX_REJECTED_LENGTH
- *                          - #SHADOW_SUFFIX_DELTA_LENGTH
- *                          - #SHADOW_SUFFIX_DOCUMENTS_LENGTH
- * @param[thingNameLength] Length of the thingName excluding the ending NULL.
+ *                       strlen( ( const char * ) myThingName ),
+ *                       strlen( ( const char * ) myShadowName ) )
+ *
+ * To use an un-named ("Classic") shadow, the shadowName length passed should be zero.
+ *
+ * @param[in] operationLength   Can be one of:
+ *                                  - #SHADOW_OP_UPDATE_LENGTH
+ *                                  - #SHADOW_OP_DELETE_LENGTH
+ *                                  - #SHADOW_OP_GET_LENGTH
+ * @param[in] suffixLength      Can be one of:
+ *                                  - #SHADOW_SUFFIX_NULL_LENGTH
+ *                                  - #SHADOW_SUFFIX_ACCEPTED_LENGTH
+ *                                  - #SHADOW_SUFFIX_REJECTED_LENGTH
+ *                                  - #SHADOW_SUFFIX_DELTA_LENGTH
+ *                                  - #SHADOW_SUFFIX_DOCUMENTS_LENGTH
+ * @param[in] thingNameLength   Length of the thingName excluding the ending NULL.
+ * @param[in] shadowNameLength  Length of the shadowName excluding the ending NULL. Zero for "Classic" shadow.
  *
  * @return Length of the shadow topic in bytes.
  */
-#define SHADOW_TOPIC_LENGTH( operationLength, suffixLength, thingNameLength ) \
-    ( operationLength + suffixLength + thingNameLength + SHADOW_PREFIX_LENGTH )
+#define SHADOW_TOPIC_LENGTH( operationLength, suffixLength, thingNameLength, shadowNameLength ) \
+    ( operationLength + suffixLength + thingNameLength + shadowNameLength +                     \
+      SHADOW_PREFIX_LENGTH +                                                                    \
+      ( ( shadowNameLength > 0 ) ? SHADOW_NAMED_ROOT_LENGTH : SHADOW_CLASSIC_ROOT_LENGTH ) )
 
 /**
- * @brief Compute the length of shadow topic "$aws/things/<thingName>/shadow/update".
+ * @brief Compute the length of shadow topic "$aws/things/<thingName>/shadow/update" or
+ *        "$aws/things/<thingName>/shadow/name/<shadowName>/update".
  */
```

Comment:
This wasn't done before, but all of these these public function-like macros should all have documentation for the the parameters. Parameter documentation being more necessary with named Shadow support.
```suggestion
 *
 * @param thingNameLength Length of thingName, excluding any possible terminating NULL.
 * @param shadowNameLength Length of shadowName, excluding possible terminating NULL. This must be zero for unnamed, classic Shadow.
 */
```

Reply:
I've done it, but just used the same documentation as already exists for SHADOW_TOPIC_LENGTH(). Similar to your suggestion, but not quite the same. Trying to not have it spill to extra lines.

I also documented the return value.

Reply:
I think this is resolved.